lib/tapioca/dsl/compilers/active_record_fixtures.rb

```diff
@@ -104,10 +109,96 @@ def method_names_from_eager_fixture_loader
 
         sig { params(mod: RBI::Scope, name: String).void }
         def create_fixture_method(mod, name)
-          mod.create_method(
-            name,
-            parameters: [create_rest_param("fixture_names", type: "T.any(String, Symbol)")],
-            return_type: "T.untyped",
+          return_type = return_type_for_fixture(name)
+          mod << RBI::Method.new(name) do |node|
+            node.add_param("fixture_name")
+            node.add_rest_param("other_fixtures")
+
+            node.add_sig do |sig|
+              sig.add_param("fixture_name", "T.any(String, Symbol)")
+              sig.add_param("other_fixtures", "NilClass")
+              sig.return_type = return_type
+            end
+
+            node.add_sig do |sig|
+              sig.add_param("fixture_name", "T.any(String, Symbol)")
+              sig.add_param("other_fixtures", "T.any(String, Symbol)")
+              sig.return_type = "T::Array[#{return_type}]"
+            end
+          end
+        end
+
+        sig { params(fixture_name: String).returns(String) }
+        def return_type_for_fixture(fixture_name)
+          model_name_from_fixture_files = fixture_file_class_mapping[fixture_name]
+          return model_name_from_fixture_files if model_name_from_fixture_files
+
+          model_name_from_fixture_set = fixture_class_name_from_fixture_set(fixture_name)
+          return model_name_from_fixture_set if model_name_from_fixture_set
+
+          model_name_from_active_record_base_class = fixture_active_record_base_class_mapping[fixture_name]
+          return model_name_from_active_record_base_class if model_name_from_active_record_base_class
+
+          "T.untyped"
```

Comment:
I wonder if this would make it easier to read:

```suggestion
          fixture_file_class_mapping[fixture_name] ||
            fixture_class_name_from_fixture_set(fixture_name) ||
            fixture_active_record_base_class_mapping[fixture_name] ||
            "T.untyped"
```

Reply:
I am very much torn on if it is better or worse to read - but I did think about it.

I'll just adopt it and rename the methods slightly so that that invocation makes more sense.

Whats the etiquette on merging here?
Squash / rebase / merge?

Reply:
No strong opinion, you can keep it as is

Reply:
after looking at it again I do prefer it, pushed

now the build failed with something something sqlite3 gem, going to rebuild